app/views/handlers/individual_response.py

```diff
@@ -491,13 +512,109 @@ def handle_post(self):
             self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
             return redirect(
                 url_for(
-                    "individual_response.get_individual_response_post_address_confirmation"
+                    "individual_response.get_individual_response_post_address_confirmation",
+                    return_to=self._return_to,
                 )
             )
 
         return redirect(
             url_for(
                 "individual_response.get_individual_response_how",
                 list_item_id=self._list_item_id,
+                return_to=self._return_to,
+            )
+        )
+
+
+class IndividualResponseWhoHandler(IndividualResponseHandler):
+    def __init__(self, schema, questionnaire_store, language, request_args, form_data):
+        self._list_name = schema.json.get("individual_response", {}).get("for_list")
+        list_model = questionnaire_store.list_store[self._list_name]
+        name_answers = []
+
+        for list_item in list_model.items:
+            if list_item != list_model.primary_person:
+                name_answers.append(
+                    (
+                        list_item,
+                        questionnaire_store.answer_store.get_answers_by_answer_id(
+                            ["first-name", "last-name"], list_item_id=list_item
+                        ),
+                    )
+                )
+
+        self.people_names = {
```

Comment:
I think we lose a bit of context with the var name, when I got to the handle_get and looking at len(self.people_names) > 1 I couldn't tell if this include primary or not. On checking back I can see its has been  removed, the name should perhaps reflect that